packages/core/src/domain/configuration.ts

```diff
@@ -69,6 +67,11 @@ export interface InitConfiguration {
   replica?: ReplicaUserConfiguration
 }
 
+export interface InitConfigurationWithDeprecated extends InitConfiguration {
```

Comment:
Instead of handling that in core by adding a new interface with deprecated keys, what about making `RumInitConfiguration` extends `Omit<InitConfiguration, 'publicApiKey', 'datacenter'>`?

Reply:
We talked about it and I think it's a bit more explicit with a specific interface, but no strong opinion. After thinking about it, in the future we may want to deprecate more options and then we could have deprecated options in both interfaces, which would be unfortunate. Using the Omit method would be less explicit but more flexible.

Reply:
Yes, personally, I found neither of these solutions, completely satisfying. But liked we discussed with Bastien, I'll go with the Omit before doing a more in depth analyses of our configuration strategy.

Reply:
IMO for the future, the configuration code should be reworked to better address our current usages. I am not super happy with the current mix of product specific options with common options in a core package.
FMU, the current need is to hide those properties from RUM users, the Omit strategy would address this need without adding extra complexity in the common configuration code.

Reply:
Do we agree that, with this approach, if a user use the package in plain JS, he will still be able to use the options ?

Reply:
what about preventing it?

Reply:
Yes, done.